exercises/Toluwalase_Agoro/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,123 @@
+#include <iostream>
+#include <memory>
+#include <iomanip>
+#include <string>
+#include "BaseShape.h"
+#include "Triangle.h"
+#include "Rectangle.h"
+#include "Square.h"
+#include "Circle.h"
+#include "ShapeSorter.h"
+
+
+
+void ShapeSorter::addShape(std::shared_ptr<BaseShape> newShape) {
+  newShape-> calculateArea();
+  newShape-> calculatePerimeter();
+  m_allShapes.emplace_back(newShape);
+
+}
+
+void ShapeSorter::displayByType(std::string& shapeType) {
+  int i{ 1 };
+  for (auto& shape : m_allShapes) {
+    if (shape->m_type == shapeType) {
+
+      shape->calculateArea();
+      shape->calculatePerimeter();
+
+      std::cout << i << ".";
+      std::cout << std::left << std::setw(10) << shape->m_type;
+      std::cout << std::setprecision(0);
+      std::cout << std::setw(20) << "Number of sides: " << shape->m_numberOfSides << "  ";
+      std::cout << std::fixed << std::setprecision(3);
+      std::cout << std::setw(8) << "Area: " << shape->m_area << "  ";
+      std::cout << std::setw(15) << "Perimeter: " << shape->m_perimeter << std::endl;
+      i++;
+    }
+  }
+}
+
+void ShapeSorter::displayByArea() {
+
+  if (m_orderToReadArea.size() == 0) {
```

Comment:
This check is not needed and could lead to bugs, for example if someone displayed the areas then added another shape and wanted to display again the new shape would not be in the list.

Reply:
Same for perimeter and number of sides